common/memops.c

```diff
@@ -146,6 +153,16 @@
 		(d) = f_round ((s) * SAMPLE_24BIT_SCALING);                    \
 	}
 
+#define float_32(s, d)                                                 \
+	if ((s) <= NORMALIZED_FLOAT_MIN) {\
+		(d) = SAMPLE_32BIT_MIN;                                        \
+	} else if ((s) >= NORMALIZED_FLOAT_MAX) {\
+		(d) = SAMPLE_32BIT_MAX;                                        \
+	} else {\
+        double extended_value = ((double)s) * SAMPLE_32BIT_SCALING;      \
```

Comment:
The temporary variable is unnecessary. The same goes for parentheses around `(double)s` because a cast takes precedence over multiplication. The macro parameter should be parenthesized on the other hand, so `(double)(s)`.

Reply:
I like the temporary variable for clarity; any compiler should optimize that away.
also like the extra parenthesis on the cast for clairty;

that said, will gladly do the changes if necessary
(and will include the parenthesize the `s` tomorrow)

Reply:
@imaami ended up rewritting quite a bit of the macro, would be thankful it if you'd review it again :)

Reply:
Hi! I happened to be looking at my phone and saw a notification, this is not my usual reaction time. :)

In other words, I'll try to keep this in mind and have a look when I can; right now I have work stuff to do.

As a side note, I noticed there's quite a lot of casting from `char *` to `int32_t *` and back. I find it unsettling as it's a potential cause of alignment bugs. And I don't mean your PR specifically, the whole file just seems a little sus.

Reply:
sounds good, just don't want to lose traction here :)
 I don't know how much we can do about the casts... the ALSA driver in particular forces you to cast back and forth between pointer types

Reply:
Do you happen to know what guarantees, if any, ALSA gives wrt. the buffers' alignment?

Reply:
yes, alsa gives us:
a pointer to the beginning of the buffer
the number of bits between each word
the size of each word and which bits to read (in case of 24 bit words packed in 32 bits)